### PR TITLE
Bootstrap project with basic API

### PR DESCRIPTION
**Pull request by jmike**
Zod already supports [non-empty array validation](https://zod.dev/?id=nonempty). However, the generated type is incompatible with [fp-ts](https://gcanti.github.io/fp-ts/).

Specifically, fp-ts defines `NonEmptyArray` as follows...

```typescript
interface NonEmptyArrayAlt<T> extends Array<T> {
  0: T;
}
```

While, zod defines `NonEmptyArray` as follows...

```typescript
type ZodNonEmptyArray<T> = [T, ...T[]];
```

This incompatibility makes it difficult to integrate zod with fp-ts.

The `zod-non-empty-array` library addresses the incompatibility issue by exposing zod transformers that return fp-ts constructs. Instead of `zod.array(...).nonempty()` use `zod.array(...).transform(transformNonEmptyArray)`.

### CODE REVIEW

**Review by nikoskalogridis — APPROVED** [review]
Great!

File: README.md [hunk 1]
```diff
@@ -1,28 +1,84 @@
...
+```
+
+## Motivation
+
+Zod already supports [non-empty array validation](https://zod.dev/?id=nonempty). However, the generated type is incompatible with [fp-ts](https://gcanti.github.io/fp-ts/).
+
+Specifically, fp-ts defines `NonEmptyArray` as follows...
+
+```typescript
+interface NonEmptyArrayAlt<T> extends Array<T> {
```
Comment (thanoskrg) [inline README.md:55]: [question] Nit, but does `fp-ts` actually has the `Alt` suffix, or is this something ours? 

```suggestion
interface NonEmptyArray<T> extends Array<T> {
```

**Review by libekonst — APPROVED** [review]
Great one, thank you! 🙏

File: lib/transformReadonlyNonEmptyArray.test.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    expect(addIssue).toHaveBeenCalledTimes(1);
+    expect(addIssue.mock.calls[0][0]).toEqual({
+      code: zod.ZodIssueCode.custom,
+      message: 'Array must not be empty',
+      fatal: true,
+    });
+  });
+
+  test('transforms non-empty array to fp-ts NonEmptyArray type', async () => {
```
Comment (thanoskrg) [inline lib/transformReadonlyNonEmptyArray.test.ts:25]: [question] Should this be:

```suggestion
  test('transforms non-empty array to fp-ts ReadonlyNonEmptyArray type', async () => {
```

**Review by thanoskrg — APPROVED** [review]
🔝 💯

**Review by NikosTsompanides — APPROVED** [review]
Magnifique 👌🏻

**Review by antonis-zisis — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (jmike) [reply README.md:55]: This is mistake. Thanks for noticing.

Comment (jmike) [reply README.md]: Fixed [a5c2e06](https://github.com/causaly/zod-non-empty-array/pull/1/commits/a5c2e0681a5601114394ce98ed03dcd586a5b38b)

File: lib/transformReadonlyNonEmptyArray.test.ts (hunk 2 above)
Comment (jmike) [reply lib/transformReadonlyNonEmptyArray.test.ts]: Fixed https://github.com/causaly/zod-non-empty-array/pull/1/commits/84528c9202562825bda95d368f07e2b705b86ea8

**Pull request merged**